x-pack/plugins/cross_cluster_replication/public/app/components/auto_follow_pattern_form.js

```diff
@@ -58,36 +90,48 @@ export class AutoFollowPatternFormUI extends PureComponent {
     remoteClusters: PropTypes.array.isRequired,
   }
 
-  state = {
-    autoFollowPattern: this.props.autoFollowPattern
-      ? {
+  constructor(props) {
+    super(props);
+
+    const isNew = this.props.autoFollowPattern === undefined;
+
+    const autoFollowPattern = isNew
+      ? getEmptyAutoFollowPattern(this.props.remoteClusters)
+      : {
         ...this.props.autoFollowPattern,
         ...getPrefixSuffixFromFollowPattern(this.props.autoFollowPattern.followIndexPattern)
-      }
-      : getEmptyAutoFollowPattern(this.props.remoteClusters),
-    isNew: typeof this.props.autoFollowPattern === 'undefined'
+      };
+
+    this.state = {
+      autoFollowPattern,
+      fieldsErrors: validateAutoFollowPattern(autoFollowPattern),
```

Comment:
Why are we validating at this point? Do we expect ES to store invalid auto follow patterns?

Reply:
D'oh I'm an idiot! The form is initially invalid, so we need to define that state to prevent it from being submitted. 😞